### DISCUSSION TITLE
Questions about Read performance while reading collection and In-memory replica!

### DISCUSSION THREAD

**Discussion by hjlee9182** [milvus-io/milvus#30611] [Q&A and General discussion]
1. I have 3 query node. I checked the memory metric while changing the in-memory value. 
When using in-memory replica number is 2, only one query node's memory is increasing?
My guess is that the memory of two query nodes should be up, but it's not. Can I ask you to explain?
![image](https://github.com/milvus-io/milvus/assets/46337218/bd084895-de90-46c1-b33d-8c38da7daaf5)

2. When I read the collection, if I write to the b collection, the reading performance decreases, and the search latency continues to increase. I understand read is relationship with Query Node, and write is relationship with index node. So I don't know the connection between these two. Why is there a problem? 
2-1. How to solve this problem? ex. increase query node etc....
2-2. Would it be recommended and possible to read and write about the same collection at the same time? I think this would lower the read performance, so how can I solve this situation?

Thanks for reading!

**Comment by yhmo** [comment]
1. Ideally, when replica=2, the memory usage is 2X; when replica=3, the memory usage is 3X. And if all the segments are indexed, the 3 query nodes should be balanced, the memory usage should be averagely.
When replica=2, a query node's memory usage is high, this is unexpected. The reason is unknown, if you could provide the logs of the 3 query nodes then we may know the reason.


2. Continually writing into a collection will definitely affect the search performance of this collection. Assume your collection has 1 million rows, search latency is 50ms, then you inserted another 10000 entities, each search request will compute more 10000 entities, search latency will incease a couple of ms, makes sense.
New inserted data are accumulated in a growing segment. For the versions before v2.3.4, growing segment has no index, search is brute-force, which is pretty slow. With continually insertion, search latency increases obviously.
In v2.3.4, we assign an IVF index for each growing segment to reduce the search latency for continually insertion.

Continually insertion(reading and writing to the same collection) definitely increase search latency. If your collection has lots of segments, increase query nodes can increase search performance, but cannot avoid the affect of continually insertion. You can try milvus v2.3.4, but v.2.3.4 also cannot completely avoid the affect of continually insertion.
